src/carnot/funcs/builtins/json_ops.h

```diff
@@ -228,6 +228,47 @@ class PluckArrayUDF : public udf::ScalarUDF {
   }
 };
 
+class SplitUDF : public udf::ScalarUDF {
+ public:
+  StringValue Exec(FunctionContext*, StringValue in, StringValue delimiter) {
+    rapidjson::StringBuffer sb;
+    rapidjson::Writer<rapidjson::StringBuffer> writer(sb);
+    writer.StartArray();
+
+    std::vector<absl::string_view> parts = absl::StrSplit(in.data(), delimiter.data());
+
+    // Verify that delimiter was found in the input string before
+    // writing the array entries.
+    if (parts.size() > 1 || parts[0] != in.data()) {
```

Comment:
I think typically, Split would return the whole string as the first element of the array if the delimiter is not in the string.

Reply:
Not sure why I went this direction, but staying consistent with the API of other split functions is definitely the right thing. Updated.